validator/keymanager/remote-web3signer/client.go

```diff
@@ -0,0 +1,192 @@
+	maxTimeout      = 3 * time.Second
+)
+
+// client a wrapper object around web3signer APIs. API docs found here https://consensys.github.io/web3signer/web3signer-eth2.html.
+type client struct {
+	BasePath   string
+	restClient *http.Client
+}
+
+// newClient method instantiates a new client object.
+//nolint:unused,deadcode
+func newClient(endpoint string) (*client, error) {
+	u, err := url.Parse(endpoint)
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid format, unable to parse url")
+	}
+	return &client{
+		BasePath: u.Host,
+		restClient: &http.Client{
+			Timeout: maxTimeout,
+		},
+	}, nil
+}
+
+// SignRequest is a request object for web3signer sign api.
+type SignRequest struct {
+	Type            string           `json:"type"`
+	ForkInfo        *ForkInfo        `json:"fork_info"`
+	SigningRoot     string           `json:"signingRoot"`
+	AggregationSlot *AggregationSlot `json:"aggregation_slot"`
+}
+
+// ForkInfo a sub property object of the Sign request,in the future before the merge to remove the need to send the entire block body and just use the block_body_root.
+type ForkInfo struct {
+	Fork                  *Fork  `json:"fork"`
+	GenesisValidatorsRoot string `json:"genesis_validators_root"`
+}
+
+// Fork a sub property of ForkInfo.
+type Fork struct {
+	PreviousVersion string `json:"previous_version"`
+	CurrentVersion  string `json:"current_version"`
+	Epoch           string `json:"epoch"`
+}
+
+// AggregationSlot a sub property of SignRequest.
+type AggregationSlot struct {
+	Slot string `json:"slot"`
+}
+
+// signResponse the response object of the web3signer sign api.
+type signResponse struct {
+	Signature string `json:"signature"`
+}
+
+// Sign is a wrapper method around the web3signer sign api.
+func (client *client) Sign(pubKey string, request *SignRequest) (bls.Signature, error) {
+	requestPath := ethApiNamespace + pubKey
+	jsonRequest, err := json.Marshal(request)
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid format, failed to marshal json request")
+	}
+	resp, err := client.doRequest(http.MethodPost, client.BasePath+requestPath, bytes.NewBuffer(jsonRequest))
+	if err != nil {
+		return nil, err
+	} else if resp.StatusCode == 404 {
+		return nil, errors.Wrap(err, "public key not found")
+	} else if resp.StatusCode == 412 {
+		return nil, errors.Wrap(err, "signing operation failed due to slashing protection rules")
+	}
+	signResp := &signResponse{}
+	if err := client.unmarshalResponse(resp.Body, &signResp); err != nil {
+		return nil, err
+	}
+	decoded, err := decodeHex(signResp.Signature)
+	if err != nil {
+		return nil, err
+	}
+	return bls.SignatureFromBytes(decoded)
+}
+
+// GetPublicKeys is a wrapper method around the web3signer publickeys api (this may be removed in the future or moved to another location due to its usage).
+func (client *client) GetPublicKeys() ([][]byte, error) {
+	const requestPath = "/publicKeys"
+	resp, err := client.doRequest(http.MethodGet, client.BasePath+requestPath, nil)
```

Comment:
check status?

Reply:
oh these only need the 2 statuses that are checked in doRequest

Reply:
cool, i missed that :)